src/init/ClusterManager.ts

```diff
@@ -0,0 +1,184 @@
+   * Spawn all required workers.
+   */
+  public spawnWorkers(): void {
+    let counter = 0;
+    this.logger.info(`Setting up ${this.workers} workers`);
+
+    for (let i = 0; i < this.workers; i++) {
+      cluster.fork().on('message', (msg: string): void => {
+        this.logger.info(msg);
+      });
+    }
+
+    cluster.on('online', (worker: Worker): void => {
+      this.logger.info(`Worker ${worker.process.pid} is listening`);
+      counter += 1;
+      if (counter === this.workers) {
+        this.logger.info(`All ${this.workers} requested workers have been started.`);
+      }
+    });
+
+    cluster.on('exit', (worker: Worker, code: number, signal: string): void => {
+      this.logger.warn(`Worker ${worker.process.pid} died with code ${code} and signal ${signal}`);
+      this.logger.warn('Starting a new worker');
+      cluster.fork().on('message', (msg: string): void => {
+        this.logger.info(msg);
+      });
+    });
+  }
+
+  /**
+   * Check whether the CSS server was booted in single threaded mode.
+   * @returns True is single threaded.
+   */
+  public isSingleThreaded(): boolean {
+    return this.clusterMode === ClusterMode.singleThreaded;
+  }
+
+  /**
+   * Whether the calling process is the primary process.
+   * @returns True if primary
+   */
+  public isPrimary(): boolean {
+    return cluster.isMaster;
+  }
+
+  /**
+   * Whether the calling process is a worker process.
+   * @returns True if worker
+   */
+  public isWorker(): boolean {
+    return cluster.isWorker;
+  }
+
+  /**
+   * Will throw an error if there are components instantiated implementing the {@link SingleThreaded}
+   * interface while the CSS is being run in multithreaded mode.
+   * @param componentsManager - The componentsManager being used to set up the application
+   */
+  public errorOnSingleThreadedComponents(componentsManager: ComponentsManager<any>): void {
+    const componentType = this.toComponentsJsType(componentsManager, 'SingleThreaded');
+    const violatingClasses: string[] = [];
+    for (const component of Object.values(componentsManager.componentResources)) {
+      if (component.isA(componentType) && component.value !== componentType) {
+        violatingClasses.push(component.value.split('#')?.[1]);
+      }
+    }
+    if (violatingClasses.length > 0) {
+      throw new InternalServerError(
+        `[${violatingClasses.join(',')}] is not threadsafe and should not be run in multithreaded setups!`,
+      );
+    }
+  }
+
+  private getFileName(): string {
+    return __filename;
+  }
+
+  /**
+   * Convert an exported interface name to the properly expected Components.js type URI.
+   * @param componentsManager - The currently used ComponentsManager
+   * @param localInterfaceName - An interface begin exported from the same source file as {@link ClusterManager}
+   * @returns A Components.js type URI
+   */
+  private toComponentsJsType(componentsManager: ComponentsManager<any>, localInterfaceName: string): string {
```

Comment:
I would remove the `localInterfaceName` parameter since we don't really need it right now and it might give wrong ideas about the function. Or it should take another (filepath?) parameter as input so it works for everything.